### PR TITLE
Feature: Update window title on page change

### PR DESCRIPTION
**Pull request by andrico1234**
Closes: https://github.com/marmelab/react-admin/issues/5893

Updates the window title to match with the title of the current page. This solves an issue with the browser history simply displaying the same title for every single page.

Note: I created a new hook called `useDocumentTitle` this is a way to set the document title. I added it to the hooks folder in `ra-core` and exported it. To me it doesn't feel like it should live there. Practically it would only be used in this Title component, so I'm not sure if we should keep the function located in `ra-core`, but there aren't any hooks exported in ra-ui-materialui`. 

Saying that, if someone chooses to pass through their own Title component, they should be able to manage their document.title however they see fit, which means they should also be able to import `useDocumentTitle`.

tl;dr, `useDocumentTitle` lives in `ra-core` but i'll be happy to place it in `ra-ui-materialui` if you see it fit

### CODE REVIEW

**Review by djhi — CHANGES_REQUESTED** [review]
Nice! As this is a new feature, can you please target the `next` branch?
